### ISSUE TITLE
Mod loader file enumeration breaks in some circumstances

### ISSUE THREAD

**Issue by WamWooWam** [Rubberduckycooly/RSDKv5-Decompilation#12]
The code here seems flawed number of ways
https://github.com/Rubberduckycooly/RSDKv5-Decompilation/blob/master/RSDKv5/RSDK/Core/ModAPI.cpp#L258-L287

- It can't handle file paths above 128 characters without smashing the stack (Windows paths can be up-to 260 characters, macOS and Linux can go as high as 4k)
- If the user has `Data/`, `Videos/` or `Bytecode/` anywhere in the path to the loader, the substring code breaks

Would it be possible to use `std::string` for the entire process, and just remove the length of the path to the mod folder from the start of the path instead? Rather than trying to support specific directories.

**Comment by WamWooWam** [comment]
While doing a test port to UWP (which *very helpfully* puts all app files in `C:\Data\...`) I changed the code above to

```cpp
auto data_rdi = fs::recursive_directory_iterator(dataPath, fs::directory_options::follow_directory_symlink);
for (auto data_de : data_rdi) {
    if (data_de.is_regular_file()) {
        std::string folderPath = data_de.path().string().substr(dataPath.string().length() + 1);
        std::transform(folderPath.begin(), folderPath.end(), folderPath.begin(),
                        [](unsigned char c) { return c == '\\' ? '/' : std::tolower(c); });
        info->fileMap.insert(std::pair<std::string, std::string>(folderPath, data_de.path().string()));
    }
}
```

I don't know if this has major issues or not, so I'm not opening a PR to replace this outright, but I thought this may be helpful to add

**Comment by MGRich** [comment]
Go ahead and open a PR; I see no issue with this code.

**Comment by Rubberduckycooly** [comment]
agreed, open a PR and it will likely be accepted asap.

**Comment by Rubberduckycooly** [comment]
implemented as of d3f2ada03abe9c9e06bcd928f5bebef45ded3d5e
